book/duckietown-robotics-development/02_basic_development/05_creating_docker_containers.md

```diff
@@ -154,45 +154,49 @@ Now that you know your way around Dockerfiles, it is time to finally build somet
 
 We will divide the image that the camera acquires into `N_SPLITS` equal horizontal sectors. `N_SPLITS` will be an environment variable we pass to the container. Think of it as a configuration parameter. The container should find which color is most present in each sector. Or alternatively you can look at the color distribution for each split. It should print the result in a nicely formatted way with a frequency of about 1Hz.
 
-You can start your Dockerfile from `duckietown/dt-duckiebot-interface:daffy-arm32v7`. Most of the stuff you need should already be in there. Make a `requirements.txt` file where you list all your pip dependencies. We would expect that you would need at least `picamera` and `numpy`. Using a `requirements.txt` file is a good practice, especially when you work with big projects. The Dockerfile then copies this file and passes it to pip which installs all the packages you specify there. Finally copy your code in the container and specify it should be the starting command. Here’s an example Dockerfile. Make sure you understand what each single line is doing. Keep in mind that you might need to modify it in order to work for you:
+You can start your Dockerfile from `duckietown/dt-duckiebot-interface:daffy-arm32v7`. Most of the stuff you need should already be in there. Make a `requirements.txt` file where you list all your pip dependencies. We would expect that you would need at least `picamera` and `numpy`. Using a `requirements.txt` file is a good practice, especially when you work with big projects. The Dockerfile then copies this file and passes it to pip which installs all the packages you specify there. Finally copy your code in the container and specify it should be the starting command. Here’s an example Dockerfile: 
 
 ```Dockerfile
 FROM duckietown/dt-duckiebot-interface:daffy-arm32v7
 
-WORKDIR /color_detector
+WORKDIR /color_detector_dir
 
 COPY requirements.txt ./
 
 RUN pip install -r requirements.txt
 
 COPY color_detector.py .
 
-CMD python ./color_detector.py
+CMD python3 ./color_detector.py
 ```
 
-Working with `picamera` can sometimes be tricky so you can use this template for `color_detector.py` to get started:
+Make sure you understand what each single line is doing. Keep in mind that you might need to modify it in order to work for you.
+
+Note: If you are using a DB-Beta, so a NVIDIA Jetson Nano powered Duckiebot, you will need to change the "dt-duckiebot-interface:daffy-arm32v7" tag to "dt-duckiebot-interface:jetson".
+
+Working with the camera can sometimes be tricky so you can use this template for `color_detector.py` to get started:
 
 ```python
-import picamera
-import picamera.array
+#!/usr/bin/env python3
+import cv2
+import numpy as np
 from time import sleep
 
-with picamera.PiCamera() as camera:
-    camera.resolution = (320, 240)
+cap = cv2.VideoCapture(2)
```

Comment:
Make sure you keep things backward compatible, this would no longer work on the Raspberry Pis. Perhaps add two tempaltes, one for each setting, or add the jetson version for the affected lines as commented under the raspberry version and explictly say (also in comment) that they should uncomment that if working on the Jetson

Reply:
I just did that without reading this